plugin/pkg/admission/noderestriction/admission.go

```diff
@@ -655,3 +661,27 @@ func (p *Plugin) admitResourceSlice(nodeName string, a admission.Attributes) err
 
 	return nil
 }
+
+func (p *Plugin) admitCSR(nodeName string, a admission.Attributes) error {
+	// The create request must come from a node with the same name in the CN of the CSR.
```

Comment:
> The create request must come from a node with the same name in the CN of the CSR.

Perhaps also mention that this only applies to node client CSRs whose CN begins with "system:node:".